eclair-core/src/main/scala/fr/acinq/eclair/db/pg/PgPaymentsDb.scala

```diff
@@ -401,5 +398,32 @@ class PgPaymentsDb(implicit ds: DataSource, lock: PgLock) extends PaymentsDb wit
     }
   }
 
+  override def removeIncomingPayment(paymentHash: ByteVector32): Unit = withMetrics("payments/remove-incoming", DbBackends.Postgres) {
+    withLock { pg =>
+      getIncomingPaymentInternal(pg, paymentHash) match {
+        case Some(incomingPayment) =>
+          incomingPayment.status match {
+            case _: IncomingPaymentStatus.Received =>
+              throw new IllegalArgumentException("Cannot remove a received incoming payment")
+            case _: IncomingPaymentStatus =>
+              using(pg.prepareStatement("DELETE FROM payments.received WHERE payment_hash = ?")) { delete =>
+                delete.setString(1, paymentHash.toHex)
+                delete.executeUpdate()
+              }
+          }
+        case None =>
+          throw new IllegalArgumentException("Unknown incoming payment")
```

Comment:
I'm not sure we should throw here, I'd rather have this method be idempotent, it's much better.
If the payment doesn't exist in the DB, I would just return (it's conceptually a success, the payment isn't in the DB anymore).